jwst/pipeline/calwebb_ami3.py

```diff
@@ -97,9 +97,13 @@ def process(self, input):
             if self.save_averages:
                 psf_avg.meta.asn.pool_name = asn['asn_pool']
                 psf_avg.meta.asn.table_name = asn.filename
-                output_file = mk_prodname(self.output_dir, prod['psf_name'], 'amiavg')
-                self.log.info('Saving averaged PSF results to %s', output_file)
-                psf_avg.save(output_file)
+                psf_avg_output_file = mk_prodname(self.output_dir, prod['psf_name'], 'amiavg')
+                self.log.info('Saving averaged PSF results to %s', psf_avg_output_file)
+                # Perform blending of metadata for all inputs to this output file
+                self.log.info('Blending metadata for averaged target {}'.format(psf_avg_output_file))
+                blend.blendfitsdata(psf_files, psf_avg)
+
+                psf_avg.save(psf_avg_output_file)
```

Comment:
So you're able now to run blend directly on the data model, before it's been saved to a FITS file? In the past the FITS file had to be created first and then run blend on it.

Reply:
This functionality was added several months ago by @stsci-hack